src/main.ts

```diff
@@ -0,0 +1,50 @@
+import { App } from './app';
+import { Action } from './action';
+import { Local } from './action/local';
+import { Github } from './action/github';
+import { SPACES_TO_INDENT_JSON } from './json';
+
+let action: Action = new Local();
+
+if (process.env.GITHUB_ACTION !== undefined) {
+    action = new Github();
+}
+
+const app = new App(action);
+
+if (!app.sanityChecksPassing()) {
+    process.exit(1);
+}
+
+/* eslint-disable-next-line no-magic-numbers */
+const FIRST_TWO_ARGUMENTS = 2;
```

Comment:
Why make this a constant? As far as I can see this is only used once (and the value is in the name which suggests this wouldn't be changing). It does make it more descriptive as to what's happening, but that could be accomplished with just a comment.

Reply:
Because a constant reflects the meaning of it. When reading `arguments.slice(2)`, thats not as verbose as `arguments.slice(FIRST_TWO_ARGUMENTS)`.

Can remove this, don't have a strong feeling for this tho.

Reply:
I understand, I just would have used a comment if you felt it was confusing.

```
const FIRST_TWO_ARGUMENTS = 2;
const config = app.createConfiguration(process.argv.slice(FIRST_TWO_ARGUMENTS));
```

vs

```
// Create configuration using the first 2 CLI arguments
const config = app.createConfiguration(process.argv.slice(2));
```

These are functionally equivalent though, so happy to go with whatever you prefer.